### PR TITLE
Create ODbL as license option

### PR DESCRIPTION
**Pull request by smathermather**
Adding ODbL as a license option to expand the license options specifically relevant to data. CC0 recently got added, so per http://opendatacommons.org/guide/, I thought adding a _share-alike_ would complete the set.

### CODE REVIEW

**Comment by benbalter** [comment]
@smathermather thanks for putting the pull request together. Mind adding a brief description and making the case for what it should be added? Makes things easier for the community to review and discuss. 

/cc @jlord

**Comment by smathermather** [comment]
No problem.  Added it in to the comments on the original pull request.

**Comment by smathermather** [comment]
Any updates on this, @benbalter? (cc @jlord). Nice to see the CC0 in here already.

**Comment by benbalter** [comment]
@konklone, @joshdata, @waldoj any strong opinions on ODBL?

**Comment by waldoj** [comment]
I confess I've never even _heard_ of ODbL, so I'm afraid I don't have any knowledge to draw on. I'll be reading about it now, though—it looks like an interesting new licensing option.

**Comment by konklone** [comment]
I know @tmcw and @lxbarth have feelings about the _merits_ of share-alike licenses. Whether ODbL is a good license at establishing share-alike if you do want it, I'm not sure. I've also heard @sbma44 say things about ODbL being questionably enforceable in the US. I don't have any expertise around it myself though!

**Comment by tmcw** [comment]
I have many strong, emotional, and very negative feelings about the ODbL. It's barely enforceable, poorly worded, rarely used in the wild, and usually the wrong thing for open data projects.

If this project only wants to present trusted licenses, it's important to note that ODbL doesn't have the legal & historical background of any of the other choices, being essentially tailor-made for [OpenStreetMap](http://www.openstreetmap.org/) and never tested in court. Its wiggle-room words are wiggly enough to merit [Community Guidelines](http://wiki.openstreetmap.org/wiki/Open_Data_License/Community_Guidelines) that are arbitrarily set and must be read alongside the license but have no legal weight.

Basically: if the goal of this project is to collect all licenses, it's worth including. If the project aims to recommend licenses, there's no strong reason to recommend the ODbL over others like the tried-and-true CC-BY-SA.

**Comment by benbalter** [comment]
FWIW, From the [goals section](https://github.com/github/choosealicense.com#immediate-goals):

> Not comprehensive. Seems like an odd goal, but there are a bajillion licenses out there. We're going to have to filter that down to a small list of those that matter.

**Comment by sbma44** [comment]
Yup, IANAL but there are substantial differences between the US and EU when it comes to the copyrightability of databases. See http://www.bitlaw.com/copyright/database.html for a discussion. If this leaves you scratching your head, rest assured that this is the correct reaction: a lot of this seems to fall in "it'll get worked out in court" territory.

Note especially that, unlike with code or other copyrighted works, US database copyrights don't necessarily attach to component portions of the database. So what if you wind up using part of someone's database (how much?), or transforming it? It's very hard to say what will happen.

In general I think it's a mistake to offer legal advice about database licensing at all, particularly in the US -- the law is incomplete, nuts and confusing. And implementing attribution clauses for data instead of code?! I mean, c'mon. Forget about it. People should just apply licenses to code, text and images; use CC0 for open databases; and try not to sue each other as much as possible.

**Comment by JoshData** [comment]
(I don't have anything to add, but thanks for the CC. This has been super informative.)

**Comment by lxbarth** [comment]
ODbL was pretty much tailored for OpenStreetMap. It is so vague (keyword "substantial extracts") that it de facto requires separate guidelines alongside it. As @tmcw said, in the case of OSM this meant community guidelines that are as long as the already lengthy license itself (4k words + 4k words). Don't add it.

**Comment by waldoj** [comment]
FWIW, I'm now pretty well convinced that this is trouble, especially with this business of community guidelines. Perhaps give the larger community time to work this license (or other database licenses, or perhaps none at all) on its own.

**Comment by benbalter** [comment]
Sounds like the prudent move would be to hold off on adding it, at least for the time being. 

It's been two weeks and the only feedback has either been "I've never heard of it" or a pretty strong :-1:.

@smathermather thanks for taking the time to put the pull request together, and hope we can revisit adding ODBL if things change.

**Comment by benbalter** [comment]
p.s.  @tmcw, @sbma44, @lxbarth et. al., thanks for taking the time to weigh in.

**Comment by smathermather** [comment]
I'm certainly not wedded to ODbL, and aware that it riles up many people, just looking for alternatives to CC0 (although very happy to see that added as an option).  Any recommendations from @tmcw, @sbma44, @lxbarth for alternatives to CC0, perhaps CC-BY-SA.

**Comment by benbalter** [comment]
@smathermather I'd be glad to merge this given a few changes:
1. Remove the `class` attribute, this isn't used
2. Remove the `category` attribute. This is only used for grouped licenses
3. Don't link to the source file in the description, which may link users outside of the site when they expect to be linked to the (in-site) license page
4. The copyright notice and first `<HR>` doesn't appear on the source license. Where did that come from?

**Comment by smathermather** [comment]
Made requested changes. Let me know if you need anything else, @benbalter.

**Comment by johanricher** [comment]
How come ODbL ([SPDX](https://spdx.org/licenses/ODbL-1.0.html), [GNU](https://www.gnu.org/licenses/license-list.en.html), [Open Definition](http://opendefinition.org/licenses/odc-odbl/)) is still not cataloged by choosealicense.com?

It's now widely used for open data, even by your standards, with [more than 2000 public repos on GitHub](https://github.com/search?q=ODbL+filename%3ALICENSE&type=Code) using ODbL.

Af far as I know, it meets the criteria listed [here](https://github.com/github/choosealicense.com/blob/gh-pages/CONTRIBUTING.md#adding-a-license), except being an OSI approved license, but such is the case of the Creative Commons licenses which are cataloged.

**Comment by smathermather** [comment]
Good point. I didn't see this didn't get merged. If WTFPL is in there, ODbL should be in there...

**Comment by sbma44** [comment]
I still think odbl’s inclusion is a poor choice. It’s an important license because OpenStreetMap is important. That doesn’t make it a *good* license or one that should proliferate.

- as discussed above it’s based on an EU-specific database right that doesn’t exist in other jurisdictions
- it’s not maintained or updated by its creators at OKF. It’s never been tested in court. Ongoing work, such as it exists, consists of LWG pronouncements that are specific to OSM. And these are usually backstopped by the contributor agreement, which simplifies away many of these problems by providing a separate covenant.
- it’s designed for databases. This project is primarily about software, right? At the risk of being blunt: the “not software?” could use significant expansion before it’s wise to include EU-specific database licenses. Open licensing for noncode copyrighted assets is a big topic—and licensing databases yet another topic. 

I worry that my skepticism will be dismissed as FUD. That’s probably unavoidable. But I’ll gently suggest that “this license annoys copyleft skeptics” doesn’t mean it’s well designed.

**Comment by smathermather** [comment]
I like the copyleft. It is an interesting and ornery approach to the world that has very important use cases, and of course, a large number of great projects that use it.

That said, I have argued since the beginning that I have no attachment to ODbL. I am not a lawyer, but trust the some of the critiques of the license above. It is a funky little artifact that has visibility due to OSM. (BTW, EU-specific database licenses have global effect, even more than EU-specific privacy regulations).

Thus, per https://github.com/github/choosealicense.com/issues/273, I have added the hidden attribute to the pull request so that it does not show up on the list of available license but can be searched for via the API.

**Comment by smathermather** [comment]
@benbalter -- can we reopen this?

**Comment by sbma44** [comment]
Thanks @smathermather—yeah, to be clear I don’t mean to start a copyleft fight—some copyleft licenses clearly belong in this project. Just describing what usually happens when I express odbl skepticism. I appreciate your openness to this argument!

**Comment by smathermather** [comment]
Understood @sbma44. I dropped this originally when it was clear it shouldn't be in the list of licenses, but I do think it's valuable that github and gitlab be able to catalog when this is used. It certainly is important enough.

**Comment by mlinksva** [comment]
ODbL-1.0 does qualify to be cataloged here. I agree with most of @sbma44's points and don't recommend its use, but there's some value in having neutral info about it available, and potentially detected by tools that consume the license dataset curated here.

@smathermather if you want to add commits to get this PR into alignment with changes here in the past 3 years, I'd merge it. (Make sure you start by pulling from master so you get the current tests.)

Check out the documentation and other records in `_licenses` and make the tests pass.

In addition to the slight frontmatter changes, I'd like to make sure the license text matches the text cataloged by SPDX, reflected in their plain text output https://github.com/spdx/license-list-data/blob/master/text/ODbL-1.0.txt If there are errors there, please fix them in https://github.com/spdx/license-list-XML (eventually we're going to use that as upstream source of license texts).

**Comment by smathermather** [comment]
@mlinksva -- I have made all the changes needed (I think) except updating the text at [spdx/license-list-XML](https://github.com/spdx/license-list-XML) to match the line-wrapping requirements. ~~But it appears the build process has stopped so I cannot verify that everything else is in line (last build was 20 hours ago).~~ Thanks!

Update: Builds are working again. I will review and edit.

File: _licenses/odbl-1.0.txt
Comment (mlinksva) [inline _licenses/odbl-1.0.txt:2]:
```suggestion
title: ODC Open Database License v1.0
```

In order to match SPDX and have test pass.

File: _licenses/odbl-1.0.txt
Comment (mlinksva) [inline _licenses/odbl-1.0.txt:5]:
```suggestion
```

This is not needed.

File: _licenses/odbl-1.0.txt
Comment (mlinksva) [inline _licenses/odbl-1.0.txt:8]:
```suggestion
```

Field no longer used.

File: _licenses/odbl-1.0.txt
Comment (mlinksva) [inline _licenses/odbl-1.0.txt:6]:
```suggestion
```

Default so shouldn't be specified.

File: _licenses/odbl-1.0.txt
Comment (mlinksva) [inline _licenses/odbl-1.0.txt:19]: these should be alpha sorted

File: _licenses/odbl-1.0.txt
Comment (mlinksva) [inline _licenses/odbl-1.0.txt:25]: these aren't the right attributes eg there is no "keep open"; look at the README for supported attributes

File: _licenses/odbl-1.0.txt
Comment (mlinksva) [inline _licenses/odbl-1.0.txt:36]: There is no copyright line in the ODbL text. It should match as closely as possible https://github.com/spdx/license-list-data/blob/c6ac269b163b65ff8ff11f1f4f385f9631dac518/text/ODbL-1.0.txt in fact copying that exact text would be good.

File: _licenses/odbl-1.0.txt
Comment (mlinksva) [inline _licenses/odbl-1.0.txt]: It appears there's a duplicate copy of the license below.

**Review by mlinksva — CHANGES_REQUESTED** [review]

**Comment by johanricher** [comment]
@smathermather Can you implement the last pending modifications suggested by @mlinksva? Let me know if you can't and I'll try to help.

**Comment by smathermather** [comment]
> @smathermather Can you implement the last pending modifications suggested by @mlinksva? Let me know if you can't and I'll try to help.

Thanks @johanricher -- I could use help. I've been traveling for a couple months and I am up to my neck in things to catch up on.

**Comment by mlinksva** [comment]
@smathermather could you enable "allow edits from maintainers" on this PR? See https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests

**Comment by smathermather** [comment]
> @smathermather could you enable "allow edits from maintainers" on this PR? See https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests

Done. Thanks.

**Comment by mlinksva** [comment]
Apologies for delayed followup. @johanricher probably the easiest thing to do is for you to make a new PR, starting with the work done so far in this branch. Then you can implement any further changes that are needed for all the tests to pass.

**Comment by johanricher** [comment]
> Apologies for delayed followup. @johanricher probably the easiest thing to do is for you to make a new PR, starting with the work done so far in this branch. Then you can implement any further changes that are needed for all the tests to pass.

I followed upon :point_right: #656. Thanks for your work!

**Comment by mlinksva** [comment]
Thanks @johanricher. I'll go ahead and close this now and take a look at #656 soon.

**Pull request closed without merging**
